src/Config/ConfigOverrides.php

```diff
@@ -70,10 +120,109 @@ public function loadOverrides($names) {
+  protected function setLockupOverrides(array $names, array &$overrides) {
+
+    // Avoid circular loops.
+    if (!$this->configFactory || in_array('system.theme', $names)) {
+      return;
+    }
+
+    // Validate we are working with the info we need.
+    $theme_info = $this->configFactory->get('system.theme');
+    if (!$this->configPagesLoader || !$theme_info || !in_array($theme_info->get('default') . '.settings', $names)) {
+      return;
+    }
+
+    // Get the default theme from the config.
+    $theme_name = $theme_info->get('default');
+    $config_page = $this->configPagesLoader->load('lockup_settings');
+
+    // Failed to load the config page or not enabled.
+    if (!$config_page || $config_page->get('su_lockup_enabled')->getString() == "1") {
+      return;
+    }
+
+    // Do the overrides.
+    $this->setLockupTextOverrides($overrides, $theme_name, $config_page);
+
+    // Get and set a file path that is relative to the site base dir.
+    $this->setLockupFileOverrides($overrides, $theme_name, $config_page);
+
+  }
+
+  /**
+   * Set the lockup text overrides.
+   *
+   * @param array $overrides
+   *   The array of overrides.
+   * @param string $theme_name
+   *   The name of the default theme.
+   * @param \Drupal\config_pages\ConfigPagesInterface $config_page
+   *   A config page object.
+   */
+  protected function setLockupTextOverrides(array &$overrides, $theme_name, ConfigPagesInterface $config_page) {
+    $overrides[$theme_name . '.settings']['lockup']['option'] = $config_page->get('su_lockup_options')->getString();
+    $overrides[$theme_name . '.settings']['lockup']['line1'] = $config_page->get('su_line_1')->getString();
+    $overrides[$theme_name . '.settings']['lockup']['line2'] = $config_page->get('su_line_2')->getString();
+    $overrides[$theme_name . '.settings']['lockup']['line3'] = $config_page->get('su_line_3')->getString();
+    $overrides[$theme_name . '.settings']['lockup']['line4'] = $config_page->get('su_line_4')->getString();
+    $overrides[$theme_name . '.settings']['lockup']['line5'] = $config_page->get('su_line_5')->getString();
+    $overrides[$theme_name . '.settings']['use_logo'] = ($config_page->get('su_use_theme_logo')->getString() == "1") ? TRUE : FALSE;
+    $overrides[$theme_name . '.settings']['logo']['use_default'] = $overrides[$theme_name . '.settings']['use_logo'];
+  }
+
+  /**
+   * Set the lockup file/logo overrides.
+   *
+   * @param array $overrides
+   *   The array of overrides.
+   * @param string $theme_name
+   *   The name of the default theme.
+   * @param \Drupal\config_pages\ConfigPagesInterface $config_page
+   *   A config page object.
+   */
+  protected function setLockupFileOverrides(array &$overrides, $theme_name, ConfigPagesInterface $config_page) {
+    $file_field = $config_page->get('su_upload_logo_image')->first();
+
+    if (!$file_field) {
+      return;
+    }
+
+    $fid = $file_field->getValue()['target_id'];
+    if (!$fid) {
+      return;
+    }
```

Comment:
if you use the config pages loader service, theres a method that takes optional parameters for the delta and the column data you want to get. it makes all this checking alot simpler. or you could do 
```
$value = $config_page->get('su_upload_logo_image')->getValue();
if (!empty($value[0]['target_id'])) {
```